zfit/core/constraint.py

```diff
@@ -124,7 +124,6 @@ def __init__(self, params: Dict[str, ZfitParameter], distribution: tfd.Distribut
         self._distribution = distribution
         self.dist_params = dist_params
         self.dist_kwargs = dist_kwargs if dist_kwargs is not None else {}
-        self._tparams = tf.stack(self.get_params())
```

Comment:
So this was the source of the problem, so `get_params` doesn't conserve the order ?

Reply:
Yes, not insertion order at least, not guaranteed...